src/cdk-experimental/menu/menu-item.ts

```diff
@@ -50,9 +53,51 @@ export class CdkMenuItem {
   }
   private _checked = false;
 
+  /**  Whether the CdkMenuItem is disabled - defaults to false */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+  private _disabled = false;
+
   /** Emits when the attached submenu is opened */
   @Output() opened: EventEmitter<void> = new EventEmitter();
 
+  constructor(
+    /** reference a parent CdkMenuGroup component */
+    private readonly _menuGroup: CdkMenuGroup
+  ) {}
+
+  /** Configure event subscriptions */
+  ngAfterContentInit() {
+    if (this.role !== 'menuitem') {
```

Comment:
Rather than using conditionals to check for the type of the menu item throughout this PR, what do you think of having separate subclasses for `menuitemradio` and `menuitemcheckbox`? I have a hunch it will make the code simpler. They might even share a base class (or interface?) like `MenuItemWithSelection`.

Reply:
Something like, CdkMenuItem, CdkMenuItemTrigger, CdkMenuItemRadio, CdkMenuItemCheckbox with CdkMenuItem being the base class and having a MenuItem interface?

Reply:
I was imagining one of these options:
![menu-item](https://user-images.githubusercontent.com/838736/84715762-4f2a1f00-af26-11ea-9613-180d9655e42f.png)

The regular menu item isn't necessarily a trigger for another menu, it can just be a regular action, so the trigger would be a separate directive that's applied to a menu item. For the checkbox and radio items, they _probably_ share some behavior, but I'm not sure how much. If it's a lot, I would add a base class for that shared code. If it's not a lot of code, then they would just extend `CdkMenuItem` directly.